File: Typescript/ways/SubstitutionCipher.ts


export class SubstitutionCipher{
    private static alphabet : string ="abcdefghijklmnopqrstuvwxyz"+"abcdefghijklmnopqrstuvwxyz ".toUpperCase();
    private static digits : string = '1234567890';
    public static ALPHABETS_DIGITS = this.alphabet+this.digits;

    public static encrypt(message : string, key:string) :string {
        let ciphertext  = '';
        for(let char_ of message){
            if(this.ALPHABETS_DIGITS.includes(char_)){
                let index = this.ALPHABETS_DIGITS.indexOf(char_)
                let newCharFrKey = key.charAt(index)
                ciphertext += newCharFrKey;
            } else {
                ciphertext+=char_;
            }
        }
        return ciphertext;
    } 

    public static decrypt(ciphertext : string, key:string) : string {
        let plaintext = "";
        for(let char_ of ciphertext){
            if (key.includes(char_)){
                let index = key.indexOf(char_);
                let newCharFrKey = this.ALPHABETS_DIGITS.charAt(index);
                plaintext += newCharFrKey;
            } else {
                plaintext += char_;
            }
        }
        return plaintext;
    }
    
}

File: Python/ways/SubstitutionCipher.py

class SubstitutionCipher:

    def __init__(self) -> None:
        self.__alphabets= 'abcdefghijklmnopqrstuvwxyz'+"abcdefghijklmnopqrstuvwxyz".upper()
        self.__digits = "1234567890"
        self.alphabets_digits = self.__alphabets + self.__digits
        self.fixed_length_key = len(self.alphabets_digits)
        pass

    def encrypt_message(self,plaintext:str, key:str):
        ciphertext = ''
        for char in plaintext:
            if char in self.alphabets_digits:
                index = self.alphabets_digits.index(char)
                newCharFromKey = key[index]
                ciphertext += newCharFromKey
            else:
                ciphertext += char
        return ciphertext

    def decrypt_message(self,ciphertext:str, key:str):
        plaintext = ''
        for char in ciphertext:
            if char in key:
                index = key.index(char)
                newCharFromKey = self.alphabets_digits[index]
                plaintext += newCharFromKey
            else:
                plaintext += char
        return plaintext
    